### ISSUE TITLE
how to train by GPU?

### ISSUE THREAD

**Issue by js418** [BrikerMan/Kashgari#6]

**Comment by BrikerMan** [comment]
just uninstall tensorflow then install tensorflow-gpu
```bash
pip uninstall tensorflow
pip install tensorflow-gpu
```

**Comment by js418** [comment]
I did uninstall tensorflow and installed tensorflow-gpu, but it seems not work

**Comment by BrikerMan** [comment]
Please provide the detailed log after run `model.fit`

**Comment by js418** [comment]
2019-02-07 18:45:29.343697: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled 
to use: AVX2 FMA

__________________________________________________________________________________________________
Layer (type)                    Output Shape         Param #     Connected to                     
==================================================================================================
Input-Token (InputLayer)        (None, 512)          0                                            
__________________________________________________________________________________________________
Input-Segment (InputLayer)      (None, 512)          0                                            
__________________________________________________________________________________________________
Embedding-Token (TokenEmbedding [(None, 512, 768), ( 16226304    Input-Token[0][0]                
__________________________________________________________________________________________________
Embedding-Segment (Embedding)   (None, 512, 768)     1536        Input-Segment[0][0]              
__________________________________________________________________________________________________
Embedding-Token-Segment (Add)   (None, 512, 768)     0           Embedding-Token[0][0]            
                                                                 Embedding-Segment[0][0]          
__________________________________________________________________________________________________
Embedding-Position (PositionEmb (None, 512, 768)     393216      Embedding-Token-Segment[0][0]    
__________________________________________________________________________________________________
Embedding-Dropout (Dropout)     (None, 512, 768)     0           Embedding-Position[0][0]         
__________________________________________________________________________________________________
Embedding-Norm (LayerNormalizat (None, 512, 768)     1536        Embedding-Dropout[0][0]          
__________________________________________________________________________________________________
Encoder-1-MultiHeadSelfAttentio (None, 512, 768)     2362368     Embedding-Norm[0][0]             
__________________________________________________________________________________________________
Encoder-1-MultiHeadSelfAttentio (None, 512, 768)     0           Encoder-1-MultiHeadSelfAttention[
__________________________________________________________________________________________________
Encoder-1-MultiHeadSelfAttentio (None, 512, 768)     0           Embedding-Norm[0][0]             
                                                                 Encoder-1-MultiHeadSelfAttention-
__________________________________________________________________________________________________
Encoder-1-MultiHeadSelfAttentio (None, 512, 768)     1536        Encoder-1-MultiHeadSelfAttention-
__________________________________________________________________________________________________
Encoder-1-FeedForward (FeedForw (None, 512, 768)     4722432     Encoder-1-MultiHeadSelfAttention-
__________________________________________________________________________________________________
......

Encoder-12-FeedForward (FeedFor (None, 512, 768)     4722432     Encoder-12-MultiHeadSelfAttention
__________________________________________________________________________________________________
Encoder-12-FeedForward-Dropout  (None, 512, 768)     0           Encoder-12-FeedForward[0][0]     
__________________________________________________________________________________________________
Encoder-12-FeedForward-Add (Add (None, 512, 768)     0           Encoder-12-MultiHeadSelfAttention
                                                                 Encoder-12-FeedForward-Dropout[0]
__________________________________________________________________________________________________
Encoder-12-FeedForward-Norm (La (None, 512, 768)     1536        Encoder-12-FeedForward-Add[0][0] 
__________________________________________________________________________________________________
non_masking_layer_1 (NonMasking (None, 512, 768)     0           Encoder-12-FeedForward-Norm[0][0]
__________________________________________________________________________________________________
bidirectional_1 (Bidirectional) (None, 512, 512)     2099200     non_masking_layer_1[0][0]        
__________________________________________________________________________________________________
dense_1 (Dense)                 (None, 512, 128)     65664       bidirectional_1[0][0]            
__________________________________________________________________________________________________
crf_1 (CRF)                     (None, 512, 29)      4640        dense_1[0][0]                    
==================================================================================================
Total params: 103,846,560
Trainable params: 2,169,504
Non-trainable params: 101,677,056
__________________________________________________________________________________________________
Epoch 1/50

  1/120 [..............................] - ETA: 47:16 - loss: 2.7695 - crf_accuracy: 0.1827 - acc: 0.9451
  2/120 [..............................] - ETA: 43:50 - loss: 1.5519 - crf_accuracy: 0.5635 - acc: 0.9447
  3/120 [..............................] - ETA: 42:35 - loss: 1.1332 - crf_accuracy: 0.6937 - acc: 0.9478

**Comment by js418** [comment]
I have figure it out.Thanks~ PS: Is there any way to save checkpoint model?

**Comment by BrikerMan** [comment]
just use keras's callbacks, like 
```
tf_board_callback = keras.callbacks.TensorBoard(log_dir='./logs', update_freq=1000)


model = CNNModel()
model.fit(train_x,
          train_y,
          val_x,
          val_y,
          batch_size=100,
          fit_kwargs={'callbacks': [tf_board_callback]})
```
here is more details about callbacks http://keras.io/callbacks/
